src/compile/nodes/EachBlock.ts

```diff
@@ -19,8 +20,20 @@ function unpack_destructuring(contexts: Array<{ name: string, tail: string }>, n
 			unpack_destructuring(contexts, element, `${tail}[${i}]`);
 		});
 	} else if (node.type === 'ObjectPattern') {
+		const used_properties = [];
+
 		node.properties.forEach((property) => {
-			unpack_destructuring(contexts, property.value, `${tail}.${property.key.name}`);
+			if (property.kind === 'rest') {
+				unpack_destructuring(
+					contexts,
+					property.value,
+					`@object_without_properties(${tail}, ${JSON.stringify(used_properties)})`
```

Comment:
I'm not sure if it's appropriate to use `JSON.stringify` on the array of properties that should be excluded. Maybe a new utility like `stringify_attributes` and `stringify` should be introduced for arrays?